saas/sdk/app/auth.py

```diff
@@ -245,19 +291,49 @@ def update_user(cls, login: str, is_admin: bool, password: Tuple[str, str] = Non
                     name=record.name,
                     disabled=record.disabled,
                     hashed_password=record.hashed_password,
-                    keystore=cls._resolve_keystore(record.keystore_id, record.keystore_password)
+                    keystore=cls._resolve_keystore(record.keystore_id, record.keystore_password),
+                    login_attempts=record.login_attempts
                 )
 
             else:
                 raise AppRuntimeError("Username does not exist", details={
                     'login': login
                 })
 
+    @classmethod
+    def update_login_attempts(cls, login: str, successful: bool) -> User:
+        with cls._Session() as session:
+            # check if this username already exists
+            user = session.query(UserRecord).get(login)
+
+            if user:
+                # reset the login attempts count, if it's a successful login
+                if successful:
+                    user.login_attempts = 0
+                # increase the login attempts count, if it's a failed login attempt
+                else:
+                    user.login_attempts += 1
+                session.commit()
+
+                return User(
+                    login=user.login,
+                    name=user.name,
+                    disabled=user.disabled,
+                    hashed_password=user.hashed_password,
+                    keystore=cls._resolve_keystore(user.keystore_id, user.keystore_password),
+                    login_attempts=user.login_attempts
+                )
+            elif not user:
```

Comment:
Why `elif not user` instead of simply `else`? I think a simple `else` will do here.